drivers/spi/spi.go

```diff
@@ -0,0 +1,44 @@
+package spi
+
+import (
+	"golang.org/x/exp/io/spi"
+	"sync"
+	"golang.org/x/exp/io/spi/driver"
+	"fmt"
+)
+
+type Connection struct {
+	connection *spi.Device
+	address int
+	mutex *sync.Mutex
+}
+
+// NewConnection creates and returns a new connection to a specific
+// spi device on a bus and address
+func NewConnection(opener driver.Opener, address int) (connection *Connection) {
+	dev, err := spi.Open(opener)
+	if err != nil {
+		panic(err)
+	}
+
+	return &Connection{connection: dev, address: address, mutex: &sync.Mutex{}}
+}
+
+// TODO: Read - the exp/io/spi implementation doesn't currently support reading
```

Comment:
@deadprogram Unfortunately, life has been hectic as I'm in the process of moving, but I've added you to the repo if there's anything you particularly want to work on regarding it. Beyond that, I hope to be getting back to it in the next week or two.